src/effects/effectparameterslot.cpp

```diff
@@ -243,3 +246,42 @@ void EffectParameterSlot::slotValueChanged(double v) {
         m_pEffectParameter->setValue(v);
     }
 }
+
+QDomElement EffectParameterSlot::toXML(QDomDocument* doc) const {
+    QDomElement knobParameterElement;
+    if (m_pEffectParameter != nullptr) {
+        knobParameterElement = doc->createElement("KnobParameter");
+        XmlParse::addElement(*doc, knobParameterElement, "Id",
+                             m_pEffectParameter->id());
+        // TODO(rryan): Do smarter QVariant formatting?
+        XmlParse::addElement(*doc, knobParameterElement, "Value",
+                            QString::number(m_pControlValue->getParameter()));
+        XmlParse::addElement(*doc, knobParameterElement, "LinkType",
+                            QString::number(m_pControlLinkType->get()));
+        XmlParse::addElement(*doc, knobParameterElement, "LinkInversion",
+                            QString::number(m_pControlLinkInverse->get()));
+    }
+
+    return knobParameterElement;
+}
+
+void EffectParameterSlot::loadValuesFromXml(const QDomElement& knobParameterElement) {
+    if (m_pEffectParameter == nullptr) {
+        return;
+    }
+    if (knobParameterElement.text().isEmpty()) {
+        m_pControlValue->reset();
+        m_pControlLinkType->set(
+            static_cast<double>(m_pEffectParameter->getDefaultLinkType()));
+        m_pControlLinkInverse->set(
+            static_cast<double>(m_pEffectParameter->getDefaultLinkInversion()));
+    } else {
+        m_pControlValue->setParameter(
+            XmlParse::selectNodeDouble(knobParameterElement, "Value"));
+        m_pEffectParameter->setValue(m_pControlValue->get());
```

Comment:
setParameter does not send value change event to the owner. Please try setParamerterFrom( .. , nullptr); 
I think you have also to deal with soft takeover since controllers will be out of sync:
m_pSoftTakeover->ignore(m_pControlValue, parameter);